brd-ios/breadwallet/SignInViewController.swift

```diff
@@ -0,0 +1,168 @@
+//
+// Created by Equaleyes Solutions Ltd
+//
+
+import UIKit
+
+protocol SignInDisplayLogic: class {
+    // MARK: Display logic functions
+    
+    func displayLogin(viewModel: SignIn.LoginData.ViewModel)
+    func displayError(viewModel: GenericModels.Error.ViewModel)
+}
+
+class SignInViewController: UIViewController, SignInDisplayLogic {
+    var interactor: SignInBusinessLogic?
+    var router: (NSObjectProtocol & SignInRoutingLogic)?
+    
+    // MARK: Object lifecycle
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+        setup()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+    
+    // MARK: Setup
+    private func setup() {
+        let viewController = self
+        let interactor = SignInInteractor()
+        let presenter = SignInPresenter()
+        let router = SignInRouter()
+        viewController.interactor = interactor
+        viewController.router = router
+        interactor.presenter = presenter
+        presenter.viewController = viewController
+        router.viewController = viewController
+        router.dataStore = interactor
+    }
+    
+    // MARK: Routing
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let scene = segue.identifier {
+            let selector = NSSelectorFromString("routeTo\(scene)WithSegue:")
+            if let router = router, router.responds(to: selector) {
+                router.perform(selector, with: segue)
+            }
+        }
+    }
+    
+    // MARK: View lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let backItem = UIBarButtonItem(title: "<", style: .plain, target: self, action: #selector(backButtonTapped))
```

Comment:
this would probablly fail the design QA... is this how its done on this project?

Reply:
it is temporary